app/src/main/java/com/example/hwone/userdto/UserPreferencesDataStore.kt

```diff
@@ -0,0 +1,74 @@
+package com.example.hwone.userdto
+
+import android.content.Context
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import com.example.hwone.Constants.USER_PREFS
+import com.example.hwone.Constants.EXTRA_EMAIL
+import com.example.hwone.Constants.EXTRA_PASSWORD
+import com.example.hwone.Constants.EXTRA_REMEMBER
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+// Extension property for DataStore
+private val Context.dataStore by preferencesDataStore(USER_PREFS)
+
+class UserPreferencesDataStore(private val context: Context) : UserPreferencesInterface {
+
+    private val dataStore get() = context.dataStore
+
+    // Private function to retrieve a value from DataStore as Flow
+    private fun <T> getValue(key: Preferences.Key<T>, defaultValue: T? = null): Flow<T?> {
+        return dataStore.data
+            .map { preferences -> preferences[key] ?: defaultValue }
+    }
+
+    // Private function to save a value in DataStore
+    private suspend fun <T> saveValue(key: Preferences.Key<T>, value: T) {
+        dataStore.edit { preferences ->
+            preferences[key] = value
+        }
+    }
+
+    override suspend fun saveLoginData(email: String, password: String, rememberMe: Boolean) {
+        saveValue(stringPreferencesKey( EMAIL_KEY), email)
+        saveValue(stringPreferencesKey( PASSWORD_KEY), if (rememberMe) password else "")
+        saveValue(booleanPreferencesKey( REMEMBER_ME_KEY), rememberMe)
```

Comment:
It is very strange logic in my opinion. The function is called saveLoginData, so it should just save login data. You must control the flow of execution (if-else) directly in the code where this function is called.

Reply:
Done